smokey/src/main/java/frc/robot/commands/AllignWithTag.java

```diff
@@ -0,0 +1,83 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Charged Up - 2023
+// File: AllignWithTag.java
+// Intent: Forms a command to allign itself with a designated april tag.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.math.controller.PIDController;
+import frc.robot.subsystems.DrivetrainSubsystem;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.common.MotorUtils;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+public class AllignWithTag extends CommandBase{
+    private boolean done = false;
+    private double tagID;
+    private double velocityValue = 0.3;
+    private PIDController yPID = new PIDController(1.0, 0.0, 0.0);
+    private DrivetrainSubsystem drivetrainsubsystem = null;
+    NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+    
+    public AllignWithTag(double tagID, DrivetrainSubsystem drivetrainSubsystem){
+        this.tagID = tagID;
+
+        this.drivetrainsubsystem = drivetrainSubsystem;
+
+        addRequirements(drivetrainsubsystem);
+    }
+
+    @Override
+    public void initialize() {
+        done = false;
+        drivetrainsubsystem.drive(new ChassisSpeeds(0,0,0));
```

Comment:
Curious to know why we need to drive at zero?  Does the robot need to be at rest to start this class?

Reply:
It's not necessary but it makes it more safe in case this command interrupts another command where the robot is moving

Reply:
Net-net ... its likely ok, but I suspect we will **not** get true 'stop-the-robot-dead-in-its-tracks' state that I suspect is intended via this line of code.

Here's my more detailed thoughts ...

I suspect that a call to zero the chassis in init will be overridden by any non-zero subsequent execute method invocation.  These two calls will almost assuredly happen within the same periodic command scheduler invocation.  So this means that the init followed by execute will happen in < 1ms (honestly, I'd imagine it would happen in the micro-second or even lower timeframe).  Depending on how quickly the CAN bus delivers the command and how quickly the motor will poll for state changes on its registers or even if the CAN bus ignores conflicting/updated messaging, the motor may or may not ever get the command.  I suspect that even if it did get that command and say it was running at the time with robot inertia behind it, the robot would not actually stop before the execute command ChassisSpeed takes over from the call to zero.

As for the naming ...
Since the command is 'align with tag' it would seem to me that there is clear intent to have it stop at the end.  Considering the name, I'm just not sure about the intent for a 'maybe stop, maybe not stop' condition at the beginning as well.

Reply:
removed the line